examples/ggez/map.rs

```diff
@@ -0,0 +1,249 @@
+use std::{collections::HashMap, path::PathBuf};
+
+use ggez::{graphics::{self, spritebatch::SpriteBatch, DrawParam}, Context, GameResult};
+use tiled::{TileLayer, Tileset};
+
+pub struct MapHandler {
+    map: tiled::Map,
+    tileset_image_cache: HashMap<String, graphics::Image>,
+    pub example_animate: bool,
+}
+
+impl MapHandler {
+    pub fn new(map: tiled::Map, ctx: &mut Context) -> GameResult<Self> {
+        // load images for the map's tilesets
+        let mut tileset_image_cache = HashMap::new();
+        for ts in map.tilesets().iter() {
+            if let Some(image) = &ts.image {
+                // image path comes in as "assets/tilesheet.png"
+                // but ggez needs it like "/assets/tilesheet.png" or it will complain
+                let mut pb = PathBuf::new();
+                pb.push("/");
+                pb.push(image.source.clone());
+                
+                // load the image
+                let mut img = graphics::Image::new(ctx, pb)?;
+                img.set_filter(graphics::FilterMode::Nearest);
+
+                tileset_image_cache.insert(
+                    ts.name.clone(),
+                    img,
+                );
+            }
+        }
+
+        Ok(Self {
+            tileset_image_cache,
+            map,
+            example_animate: false,
+        })
+    }
+
+    pub fn width(&self) -> u32 {
+        self.map.width
+    }
+
+    pub fn height(&self) -> u32 {
+        self.map.height
+    }
+
+    pub fn tile_width(&self) -> u32 {
+        self.map.tile_width
+    }
+
+    pub fn tile_height(&self) -> u32 {
+        self.map.tile_height
+    }
+
+    pub fn get_bounds(&self) -> graphics::Rect {
+        graphics::Rect::new(0.0, 0.0, (self.height() * self.tile_height()) as f32, (self.height() * self.tile_height()) as f32)
+    }
+
+    pub fn background_color(&self) -> Option<graphics::Color> {
+        self.map.background_color
+            .map(|c| ggez::graphics::Color::from_rgba(c.red, c.green, c.blue, c.alpha))
+    }
+
+    pub fn draw(&mut self, ctx: &mut Context, draw_param: DrawParam, parallax_pan: (f32, f32)) -> GameResult {
+
+        // could be cached for more performance
+        let layer_batches: Vec<Vec<SpriteBatch>> = self.generate_map_render(ctx, parallax_pan);
+
+        // draw tile layers
+
+        // for each layer
+        for layer in &layer_batches {
+            // for each tileset in the layer
+            for batch in layer {
+                graphics::draw(ctx, batch, draw_param)?;
+            }
+        }
```

Comment:
Could be flattened to a single vector if layer batches are sequentially placed by index